app/Main.hs

```diff
@@ -155,13 +158,26 @@ main = do
             putStrLn $ "DDlog " ++ dDLOG_VERSION ++ " (" ++ gitHash ++ ")"
             putStrLn $ "Copyright (c) 2019-2020 VMware, Inc. (MIT License)"
         ActionValidate -> do
-            _ <- parseValidate config'
-            return ()
+            timeAction ("validating " ++ show (confDatalogFile config)) $ do
+                _ <- parseValidate config'
+                return ()
         ActionCompile -> do
-            compileProg config'
-            -- Run rustfmt on the generated code if it's enabled
-            when (confRunRustfmt config') $
-                runCommandReportingErr "rustfmt" "cargo" ["fmt", "--all"] $ Just (confOutputDir config')
+            timeAction ("compiling " ++ show (confDatalogFile config)) $ do
+                compileProg config'
+                -- Run rustfmt on the generated code if it's enabled
+                when (confRunRustfmt config') $
+                    runCommandReportingErr "rustfmt" "cargo" ["fmt", "--all"] $ Just (confOutputDir config')
+
+-- Perform IO action and measure its duration.
+timeAction :: String -> IO () -> IO ()
```

Comment:
Make the output conditional on some option (-verbose?).

Reply:
What's wrong with always printing it?  Rust compiler does this bu default, and I find it useful, also as explicit indication that compilation succeeded.

Reply:
The Unix philosophy is to print nothing on success.
If you write tools that consume the output of the compiler you may expect that there is no output if there are no errors or warnings. But this is certainly a low priority item; we can wait until someone asks for a -quiet flag.